### PR TITLE
Daint Upgrade

### PR DESCRIPTION
**Pull request by pspoerri**

### CODE REVIEW

**Comment by lxavier** [comment]
@kosterried FYI , Pascal is working on adapting the build for Kesch, sorry if you ave also started somethin gin parallel

File: env.daint.sh [hunk 1]
```diff
@@ -151,14 +148,13 @@ unsetCppEnvironment()
         exit 1
     esac
 
+    module unload CMake
+    module unload ncurses
```
Comment (lxavier) [inline env.daint.sh]: why do you need to unload this, it was not loaded, is there something wrong with the CMake load/unload

File: env.daint.sh [hunk 2]
```diff
@@ -201,17 +197,16 @@ setFortranEnvironment()
...
-        module swap cudatoolkit/7.0.28-1.0502.10742.5.1
+        module load craype-accel-nvidia60
     fi
 
     # compiler specific modules
     case "${compiler}" in
     cray )
-        module unload cce
-        module load cce/8.4.0
+#        module unload cce
```
Comment (lxavier) [inline env.daint.sh]: Is it not possible to keep a fixed cray cce version, if we don't load explicitly we will be using the default which may change wihtout explicit notice

File: env.daint.sh (hunk 1 above)
Comment (pspoerri) [inline env.daint.sh]: because it is a dependency of CMake

**Comment by pspoerri** [comment]
Fyi, we currently get a segfault with C++:
```
> srun -C gpu -n 1 ./CommunicationFrameworkUnittestCUDA
srun: job 46481 queued and waiting for resources
srun: job 46481 has been allocated resources
CommunicationUnittest

usage: CommunicationUnittest [-periodic]
[==========] Running 50 tests from 8 test cases.
[----------] Global test environment set-up.
[----------] 10 tests from HaloUpdateManagerUnittest
[ RUN      ] HaloUpdateManagerUnittest.PeriodicBoundaryConditions
srun: error: nid02718: task 0: Segmentation fault
srun: Terminating job step 46481.0
```

**Comment by pspoerri** [comment]
I had to export: 
```
export MV2_ENABLE_AFFINITY=0; export MV2_USE_CUDA=1; export MPICH_RDMA_ENABLED_CUDA=1; export  MPICH_G2G_PIPELINE=256;
```
and now it works.

**Pull request merged**